internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -428,6 +440,74 @@ func (r *ClusterDeploymentsReconciler) updateInstallConfigOverrides(ctx context.
 	return nil
 }
 
+func (r *ClusterDeploymentsReconciler) syncManifests(ctx context.Context, cluster *common.Cluster, manifests map[string]string, alreadyCreatedManifests models.ListManifests) error {
+	r.Log.Debugf("Going to sync list of given with already created manifests")
+
+	for _, manifest := range alreadyCreatedManifests {
+		if _, ok := manifests[manifest.FileName]; !ok {
+			r.Log.Infof("Deleting cluster %s manifest %s", cluster.ID, manifest.FileName)
+			_ = r.Manifests.DeleteClusterManifestInternal(ctx, operations.DeleteClusterManifestParams{
+				ClusterID: *cluster.ID,
+				FileName:  manifest.FileName,
+				Folder:    swag.String(models.ManifestFolderOpenshift),
+			})
+		} else {
+			delete(manifests, manifest.FileName)
+		}
+	}
+
+	for filename, manifest := range manifests {
+		r.Log.Infof("Creating cluster %s manifest %s", cluster.ID, filename)
+		_, err := r.Manifests.CreateClusterManifestInternal(ctx, operations.CreateClusterManifestParams{
+			ClusterID: *cluster.ID,
+			CreateManifestParams: &models.CreateManifestParams{
+				Content:  swag.String(base64.StdEncoding.EncodeToString([]byte(manifest))),
+				FileName: swag.String(filename),
+				Folder:   swag.String(models.ManifestFolderOpenshift),
+			}})
+		if err != nil {
+			r.Log.WithError(err).Errorf("Failed to create cluster %s manifest %s", cluster.ID, filename)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ClusterDeploymentsReconciler) addCustomManifests(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment,
+	cluster *common.Cluster) error {
+
+	if !common.IsInPreInstallationState(cluster) {
+		r.Log.Debugf("Cluster is not in pre installation state, skipping")
+		return nil
+	}
+
+	alreadyCreatedManifests, err := r.Manifests.ListClusterManifestsInternal(ctx, operations.ListClusterManifestsParams{
+		ClusterID: *cluster.ID,
+	})
+	if err != nil {
+		r.Log.WithError(err).Errorf("Failed to list cluster %s manifests", cluster.ID)
+		return err
+	}
+
+	if clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef == nil && len(alreadyCreatedManifests) < 1 {
+		r.Log.Debugf("Nothing to do, skipping manifest creation")
+		return nil
+	}
+
+	configuredManifests := &corev1.ConfigMap{}
+	configuredManifests.Data = map[string]string{}
+	if clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef != nil {
+		configuredManifests = &corev1.ConfigMap{}
+		err = r.Get(ctx, types.NamespacedName{Namespace: clusterDeployment.Namespace, Name: clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef.Name}, configuredManifests)
+		if err != nil {
+			r.Log.WithError(err).Errorf("Failed to get configmap %s in %s", clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef.Name, clusterDeployment.Namespace)
+			return err
+		}
+	}
```

Comment:
Consider extracting to another function and calling that function from `syncManifests` where the `configuredManifests` is actually used.

Reply:
I think such validation is required before we allow the installation to start.

Reply:
Not sure i understand.  if configmap reference was set and configmap will not be with us, we will fail and return error.
What else can we do?